src/components/user-defined-sets/CurrentSetManager.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { setsReducer, NAME_CURRENT_SET } from './sets';
+
+const CURRENT_SELECTION = 'Current selection';
+
+export default class CurrentSetManager extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isEditing: false,
+      setName: '',
+    };
+
+    this.startEditing = this.startEditing.bind(this);
+    this.cancelEditing = this.cancelEditing.bind(this);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  startEditing(event) {
+    event.preventDefault();
+    const { sets } = this.props;
+    const nextIndex = Array.from(sets.namedSets.keys()).length + 1;
+    this.setState({ isEditing: true, setName: `Set ${nextIndex}` });
+  }
+
+  cancelEditing() {
+    this.setState({ isEditing: false, setName: '' });
+  }
+
+  handleChange(event) {
+    this.setState({ setName: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { setName } = this.state;
+    const { sets, onUpdateSets } = this.props;
+    onUpdateSets(setsReducer(sets, {
+      type: NAME_CURRENT_SET, key: setName, clear: true,
+    }));
+    this.cancelEditing();
```

Comment:
Better name? I'm guessing it just leaves the editting mode... but it's not like changes are lost?